moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -566,6 +566,16 @@ bool Trajectory::integrateForward(std::list<TrajectoryStep>& trajectory, double
     trajectory.push_back(TrajectoryStep(path_pos, path_vel));
     acceleration = getMinMaxPathAcceleration(path_pos, path_vel, true);
 
+    if (path_vel == 0 && acceleration == 0)
```

Comment:
I'm generally wary of doing direct floating-point equality, but I think in this case it's probably okay since it will come directly from a parameter and isn't calculated on the fly.

Reply:
I considered comparing with epsilon, but I'm not sure if I'll be preventing trajectories that would still end but simply take longer. You're correct that, in the case that a component of max_accel==0, acceleration comes from a min() operation, so acceleration is guaranteed to be 0.